drivers/metadata/metadata.go

```diff
@@ -172,13 +172,26 @@ func NewCatalogSet(v []Catalog) *CatalogSet {
 		r[i] = &v[i]
 	}
 	return &CatalogSet{
-		resultSet: resultSet{
+		resultSet: &resultSet{
 			results: r,
 			columns: []string{"Catalog"},
 		},
 	}
 }
 
+func NewCatalogSetWithColumns(cols []string) *CatalogSet {
```

Comment:
If this would also accept the results slice, would AddResult be unnecessary?

Reply:
I like the idea of providing a custom result struct that implements `Values` but I wonder how to handle the verbose version of the describe commands, which prints additional columns. Right now this is only handled by the writer settings columns and a scan callback: https://github.com/xo/usql/blob/master/drivers/metadata/writer.go#L265

I don't think that was the best decision. It would be much simpler to mark verbose columns in an additional list in the `resultSet`. `Values()` should always return values for all columns and the writer would be responsible for filtering out unused ones. That way it can treat results as immutable.

It would be nice if you'd address this, but it's a lot to ask and it's fine to do in a follow-up PR. Let me know what you think.

Reply:
Yes, but do the for-loop conversion in the Catalogs function.

Reply:
Is that necessary?

Reply:
Wait a minute, I'll change it

Reply:
change to 

```
func NewCatalogSetWithColumns(v []Result, cols []string) *CatalogSet {
	return &CatalogSet{
		resultSet: &resultSet{
			results: v,
			columns: cols,
		},
	}
}
```

```
	var result []metadata.Result
	for rows.Next() {
		rec := Catalog{}
		err = rows.Scan(&rec.Catalog, &rec.Owner, &rec.Encoding, &rec.Catalog, &rec.Ctype, &rec.AccessPrivileges)
		if err != nil {
			return nil, err
		}
		result = append(result, &rec)
	}
	if rows.Err() != nil {
		return nil, rows.Err()
	}
	return metadata.NewCatalogSetWithColumns(result, catalogsColumnName), nil
```

AddResult is no longer required